activerecord/test/cases/associations/has_one_associations_test.rb

```diff
@@ -529,7 +529,8 @@ def test_replacement_failure_due_to_new_record_should_raise_error
 
     assert_equal "Failed to save the new associated ship.", error.message
     assert_equal ships(:black_pearl), pirate.ship
-    assert_equal pirate.id, pirate.ship.pirate_id
+    assert_nil pirate.ship.pirate_id
```

Comment:
This might be undesired. Rewinding the attributes from this update nullifies the associated ship's pirate ID. Thoughts?

Reply:
Potentially, if you call `#save` on them once they've been rolled back under very specific circumstances. I'll try to fix this.

Reply:
Actually, this turned out to be a really big problem. It would ignore any dirty changes after the last save in the transaction. I fixed it by using the current `@attributes` in `#rewind_attribute_assignments` as opposed to `@mutations_before_last_save`.